browser/ui/webui/brave_rewards_page_ui.cc

```diff
@@ -709,6 +709,10 @@ void RewardsDOMHandler::OnPromotionFinished(
       promotion.promotion_id,
       result,
       std::move(promotion_ptr));
+
+  if (promotion.type == 1) {  // ads
```

Comment:
this should be done in `RewardsServiceImpl::OnAttestPromotion`

Reply:
Moved inside lib so same business logic is shared across all platforms